### PR TITLE
fix: check for undefined or no ID

### PR DESCRIPTION
**Pull request by Danfraga33**

### CODE REVIEW

**Review by Danfraga33 — COMMENTED** [review]
Adding article types

File: src/pages/authors/index.astro [hunk 1]
```diff
@@ -4,14 +4,15 @@ import Info from "~/assets/svgs/info.astro";
...
+import type {Author} from "~/lib/types"
 import { getMeta } from "~/lib/utils/getMeta";
 const entry = await getEntry("views", "authors");
 
 if (!entry) {
   return Astro.redirect("/404");
 }
 
-const authors = authorsHandler.allAuthors();
+const authors:Author[] = authorsHandler.allAuthors();
```
Comment (Mrahmani71) [inline src/pages/authors/index.astro:15]: In this section, there's no need to define the types manually; Astro will generate them automatically

File: src/lib/handlers/articles.ts [hunk 2]
```diff
@@ -17,6 +17,12 @@ export const articlesHandler = {
 
     headlineSmall: () => {
         const bigHeadline = articlesHandler.headlineBig();
+
+        if (!bigHeadline || !bigHeadline.id) {
+			console.error('bigHeadline is undefined or missing id!');
+			return [];
+		}
+
```
Comment (Mrahmani71) [inline src/lib/handlers/articles.ts:25]: I modified this section to check if mainHeadline (or bigHeadline) exists in a different way, making this part of the code unnecessary.

File: src/lib/types/index.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export type Link = {
   icon?: string;
 }
 
-type Author = {
+export type Author = {
   name: string;
   link: string;
 }
```
Comment (Mrahmani71) [inline src/lib/types/index.ts]: We don't need this custom type for now; maybe we'll use it later.

File: src/pages/_home/latestNews.astro [hunk 4]
```diff
@@ -3,8 +3,9 @@
 import NewsCard from "~/components/cards/newsCard.astro";
 import HeaderSection from "./headerSection.astro";
 import { articlesHandler } from "~/lib/handlers/articles";
+import type { Article } from "~/lib/types";
 
-const articles = articlesHandler.allArticles();
+const articles: Article[] = articlesHandler.allArticles();
```
Comment (Mrahmani71) [inline src/pages/_home/latestNews.astro:8]: We don't need this custom type now; Astro will generate it automatically.

File: src/components/shared/view-list-header.astro [hunk 5]
```diff
@@ -1,12 +1,40 @@
...
+    return result.trim();
+}
+const headerTitle = formatString(title, categories)
 ---
 
 <section
   class="flex items-center border-b-2 border-primary/10 p-1 mb-4"
 >
-  <h1 class="text-xl font-bold capitalize">{title}</h1>
+  <h1 class="text-xl font-bold capitalize">{headerTitle}</h1>
```
Comment (Mrahmani71) [inline src/components/shared/view-list-header.astro:39]: This is a good idea, but I want to keep this project very simple and avoid adding complicated features for now.

**Review by Mrahmani71 — COMMENTED** [review]
I'm sorry for closing this pull request. The types and features you added are not necessary as Astro will generate them automatically. For now, I want to keep this project very simple. Thank you for your contributions 💐

**Pull request closed without merging**